static-checks/ansible/allowed-modules/test.py

```diff
@@ -0,0 +1,134 @@
+    if len(task_module) == 1:
+        module = str(task_module[0])
+        if modules_list.count(module) == 0:
+            # Module module is not yet in the list, append it.
+            modules_list.append(module)
+    else:
+        module = ''.join([x for x in list(updated_keywords) if x not in
+                          ansible_reserved_keywords])
+        results.report('fail', module, 'module is not allowed')
+        return 1
+    return 0
+
+
+def traverse_blocks(modules_list, task, allowed_modules):
+    block_set = set(['block'])
+    task_keywords = set(task.keys())
+
+    if len(task_keywords.intersection(block_set)) > 0:
+        for block_task in task['block']:
+            return traverse_blocks(modules_list, block_task, allowed_modules)
+    else:
+        if list_append_ansible_module_from_task(modules_list,
+                                                task_keywords,
+                                                allowed_modules):
+            util.log(f"This occurred in the following task:\n{task}\n")
+            return
+
+
+def check_allowed_modules(playbook_path, allowed_modules):
+    """
+    Collects all modules from Ansible playbook playbook_path and reports pass
+    if a module is found in the allowed_modules set, otherwise reports fail
+    and prints an offending Ansible module and task. Function also handles
+    Ansible block sections which can bundle multiple tasks.
+    Note: ansible playbook must be a valid playbook which means that
+    'ansible-playbook --syntax-check' must not report any errors for it.
+    """
+    with open(playbook_path, 'r') as yamlfile:
+        playbook = yaml.safe_load(yamlfile)
+    modules_list = []
+
+    util.log(f"Check Ansible playbook {playbook_path} contains only allowed modules")
+    for section in playbook:
+        if section['tasks'] is None:
+            # Playbook has no tasks.
+            return
+        for task in section['tasks']:
+            traverse_blocks(modules_list, task, allowed_modules)
+
+    modules_list.sort()
+    for m in modules_list:
+        results.report('pass', m, 'module is allowed')
+
+
+# rhc-worker-playbook modules, exported per official instructions on
+# https://access.redhat.com/articles/remediation
+os.environ['ANSIBLE_COLLECTIONS_PATH'] = \
+    '/usr/share/rhc-worker-playbook/ansible/collections/ansible_collections/'
+
+with select_all_rules(util.get_datastream()) as ds_file:
+    with open(ds_file) as ds:
+        with tempfile.NamedTemporaryFile(mode='w', suffix='.yml') as playbook:
+            util.log(f"Generate Ansible playbook {playbook.name} for all rules from {ds_file}")
+            oscap_cmd = ['oscap', 'xccdf', 'generate', 'fix', '--fix-type', 'ansible', '--output',
+                         playbook.name, ds.name]
+            util.subprocess_run(oscap_cmd, check=True)
+
+            ansible_cmd = ['ansible-playbook', '--syntax-check', playbook.name]
+            util.subprocess_run(ansible_cmd, check=True)
```

Comment:
A minor thing, but I would rather include the `[]` inside `util.subprocess_run()` if it fits on a single line.

In this case, it would be exactly 98 characters, which is almost the maximum allowed, but still allowed. :smile:
```python
            util.subprocess_run(['ansible-playbook', '--syntax-check', playbook.name], check=True)
```